assets/javascripts/modules/autoComplete.js

```diff
@@ -379,15 +381,11 @@ var getSuggestions = function (el) {
 
 /**
   * Find input to post value back to
+  * @param el
 */
 var getTargetInput = function (el) {
-  var target;
-  // input to post back a selection from the autocomplete
-  var dataPostTo = $('[id="' + el.attr("data-target-elem") + '"]');
-  if(dataPostTo.length > 0){
-    target = dataPostTo;
-  }
-  return target;
+  var $targets = el.first().data('target-elem') ? el.first().data('target-elem') : $targetInputElem;
```

Comment:
so `$targetInputElem` in the backwardly compatible way is the 3rd argument sent in to the `autoComplete.js` shown [here](https://github.com/hmrc/assets-frontend/blob/master/assets/javascripts/modules/autoCompleteFactory.js#L13) it is an element.
@adamliptrot-oc has added the new way of sending in a target element via a string set via the `data-target-elem`.
The way of sending in the `$targetInputElem` via the 3rd argument overrides the `data-target-elem`.
Both of these result in `jQuery` elements being passed back and both are optional and guarded as such in the code.

Unless you are placing a `jQuery` element in the `data-target-elem` which would mean querying in your service-Frontend (which you shouldn't be) I'm not sure how this is working? (unless I am missing something)